src/plugins/lightscript.js

```diff
@@ -134,40 +134,45 @@ pp.parseObjectComprehension = function(node) {
   return this.finishNode(node, "ObjectComprehension");
 };
 
-// c/p parseBlock
+pp.finishWhiteBlock = function(node) {
+  if (!node.body.length) {
+    this.unexpected(node.start, "Expected an Indent or Statement");
+  }
+
+  this.addExtra(node, "curly", false);
+  return this.finishNode(node, "BlockStatement");
+};
 
-pp.parseWhiteBlock = function (allowDirectives?, isExpression?) {
+pp.parseInlineWhiteBlock = function(node) {
+  if (this.state.type.startsExpr) return this.parseMaybeAssign();
+  // oneline statement case
+  node.body = [this.parseStatement(true)];
```

Comment:
I tried this change and there is a problem. `parseArrowFunctionBody` is doing a fixup here: https://github.com/wcjohnson/babylon-lightscript/blob/upstream/oneline-whiteblocks/src/plugins/lightscript.js#L328

Because the parser is no longer producing `BlockStatement`s after this change, functions are getting false-flagged as `expression=true` when they aren't expression bodies.